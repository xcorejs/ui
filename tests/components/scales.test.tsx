import 'jest-styled-components';

import { Box, colors, createScales, createTheme, darkColorTheme, XcoreProvider } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Scales - no theme', () => {
  const component = renderer.create(
    <XcoreProvider>
      <Box color="primary" />
    </XcoreProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Scales - light theme', () => {
  const lightTheme = createTheme();

  const component = renderer.create(
    <XcoreProvider theme={lightTheme}>
      <Box color="primary" />
    </XcoreProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Scales - dark theme', () => {
  const darkTheme = createTheme({
    ...createScales({
      ...colors(darkColorTheme)
    })
  });

  const component = renderer.create(
    <XcoreProvider theme={darkTheme}>
      <Box color="primary" />
    </XcoreProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
