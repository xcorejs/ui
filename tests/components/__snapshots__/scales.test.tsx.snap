// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scales - dark theme 1`] = `
.c0 {
  color: #DAA520;
}

<div
  className="c0"
/>
`;

exports[`Scales - light theme 1`] = `
.c0 {
  color: #0171b6;
}

<div
  className="c0"
/>
`;

exports[`Scales - no theme 1`] = `
.c0 {
  color: #0171b6;
}

<div
  className="c0"
/>
`;
